src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -0,0 +1,119 @@
+package nextstep.courses.domain;
+
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Session {
+    private Long id;
+    private String image;
+    private LectureType lectureType;
+    private LectureStatus lectureStatus;
+    private LocalDate startDate;
+    private LocalDate endDate;
+    private List<Long> userIds;
+    private int maxUser;
+
+    private Session(Builder builder) {
+        this.id = builder.id;
+        this.image = builder.image;
+        this.lectureType = builder.lectureType;
+        this.lectureStatus = builder.lectureStatus;
+        this.startDate = builder.startDate;
+        this.endDate = builder.endDate;
+        this.userIds = builder.userIds;
+        this.maxUser = builder.maxUser;
+    }
+
+    public LocalDate getStartDate() {
+        return startDate;
+    }
+
+    public LocalDate getEndDate() {
+        return endDate;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public String getImage() {
+        return image;
+    }
+
+    public LectureType getLectureType() {
+        return lectureType;
+    }
+
+    public LectureStatus getLectureStatus() {
+        return lectureStatus;
+    }
+
+    public void register(long userId) {
+        if (!this.lectureStatus.equals(LectureStatus.RECRUITING)) {
```

Comment:
```suggestion
        if (!this.lectureStatus.recruiting()) {
```
enum 객체도 값을 꺼내기 보다 메시지를 보내는 방식으로 구현해 보는 것은 어떨까?